public/app/plugins/datasource/cloudwatch/datasourcecfg.cue

```diff
@@ -0,0 +1,82 @@
+// Copyright 2023 Grafana Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grafanaplugin
+
+import (
+	common "github.com/grafana/grafana/packages/grafana-schema/src/common"
+	"github.com/grafana/grafana/pkg/plugins/pfs"
+)
+
+// This file (with its sibling .cue files) implements pfs.GrafanaPlugin
+pfs.GrafanaPlugin
+
+composableKinds: DataSourceCfg: {
+	maturity: "merged"
+
+	lineage: {
+		seqs: [
+			{
+				schemas: [
+					{
+						#LogGroup: {
+							// ARN of the log group
+							arn: string
+							// Name of the log group
+							name: string
+							// AccountId of the log group
+							accountId?: string
+							// Label of the log group
+							accountLabel?: string
+						} @cuetsy(kind="interface")
+
+						#AwsAuthType: "keys" |
+							"credentials" |
+							"default" |
+							"ec2_iam_role" |
+							// @deprecated use default
+							"arn" @cuetsy(kind="enum", memberNames="Keys|Credentials|Default|EC2IAMRole|ARN")
+
+						#AwsAuthDataSourceJsonData: {
+							common.DataSourceJsonData
+							authType?: #AwsAuthType
+							// ARN of the role to assume
+							assumeRoleArn?: string
+							externalId?:    string
+							profile?:       string
+							// Default AWS region to use
+							defaultRegion?: string
+							endpoint?:      string
+						}
+
+						Options: {
+							#AwsAuthDataSourceJsonData
```

Comment:
Question: We export a base type called [`AwsAuthDataSourceJsonData`](https://github.com/grafana/grafana-aws-sdk-react/blob/60f4c81d71a7290f29d05c12c8398991bba3863e/src/types.ts#L14-L21) from `@grafana/aws-sdk` that can be used by in AWS specific datasources. The [`CloudWatchJsonData`](https://github.com/grafana/grafana/blob/57d3027cb008e8e646357c9bb6b07947142f79a8/public/app/plugins/datasource/cloudwatch/types.ts#L35) type for CloudWatch extends `AwsAuthDataSourceJsonData`, how should we schematize this type that is common between AWS datasources?

Reply:
I think that you can create this schema inside [common](https://github.com/grafana/grafana/tree/main/packages/grafana-schema/src/common) package and import it.